ocean-client/src/vault-maxi.ts

```diff
@@ -76,17 +76,34 @@ export async function main(event: maxiEvent): Promise<Object> {
     }
 
     let vault: LoanVaultActive = vaultcheck
+
+    // 2022-03-08 Krysh: Something went wrong on last execution, we need to clean up, whatever was done
+    if (settings.state !== VaultMaxiState.Start && settings.state !== VaultMaxiState.Finish) {
```

Comment:
add blockheight to state. have state split into the txs.
basically: if waiting for a transaction and blockheight still the same: wait for next block.
after next block: 
if was waiting for first transaction : clean up -> return
if was waiting for second transaction: all good, we are ready now.